### PR TITLE
TENSOR_COLUMN_NAME removal

### PR DESCRIPTION
**Pull request by Bhav00**
## Why are these changes needed?

The changes are related to the issue https://github.com/ray-project/ray/issues/37047.

I removed the code with TENSOR_COLUMN_NAME.

Removed the whole function where parameters/variables the function was working upon depended on the TENSOR_COLUMN_NAME. Otherwise, I tried to cut only a part of the code so as to maintain the functionality of the rest even without TENSOR_COLUMN_NAME wherever possible.
Certain functions in test files were kept to not break CI break, hence the TENSOR_COLUMN_NAME was kept in the constants file. 
The requested changes in the previous Pull request were made sure to be present.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Bhav00** [comment]
@scottjlee this is the new pr.
I waited for the tests to happen through this CI pipeline before pinging.
Any other changes that you would like me to make?

**Comment by scottjlee** [comment]
Thanks @Bhav00 for creating the new PR. The changes largely LGTM, but I see there are a couple remaining Data test failures, which might be stragglers from carrying over previous changes:
```
FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_pandas[True]
--
  | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_pandas[False]
  | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_multi_dim_pandas[True]
  | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_multi_dim_pandas[False]
  | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_object_pandas
```

Could you take a stab at fixing these? At a first glance, seems to be related to some batch format issue. Thanks!

**Comment by Bhav00** [comment]
> Thanks @Bhav00 for creating the new PR. The changes largely LGTM, but I see there are a couple remaining Data test failures, which might be stragglers from carrying over previous changes:
> 
> ```
> FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_pandas[True]
> --
>   | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_pandas[False]
>   | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_multi_dim_pandas[True]
>   | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_multi_dim_pandas[False]
>   | FAILED python/ray/air/tests/test_data_batch_conversion.py::test_numpy_object_pandas
> ```
> 
> Could you take a stab at fixing these? At a first glance, seems to be related to some batch format issue. Thanks!

Apparently, these functions, `test_numpy_pandas`, `test_numpy_multi_dim_pandas` and `test_numpy_object_pandas`, all are having `TENSOR_COLUMN_NAME` in them. Because these functions are made around `TENSOR_COLUMN_NAME`, I had to remove them whole previously altogether but that also resulted in CI giving out errors so I was told to keep them.

**Comment by Bhav00** [comment]
And yet again I forgot to ping, please refer to the comment I wrote just above this.
@scottjlee

**Comment by scottjlee** [comment]
I see, but the CI is still failing because those tests are erroring right? So if I understand correctly, either we need to take steps to fully remove the tests, or fix the errors such as [this one](https://buildkite.com/ray-project/oss-ci-build-pr/builds/30703#0189cbfc-3ce9-49bc-8124-b108df356444/2016-3075):
```
___________________________ test_numpy_object_pandas ___________________________
--
  |  
  | def test_numpy_object_pandas():
  | input_data = np.array([[1, 2, 3], [1]], dtype=object)
  | expected_output = pd.DataFrame({TENSOR_COLUMN_NAME: input_data})
  | >       actual_output = _convert_batch_type_to_pandas(input_data)
  |  
  | python/ray/air/tests/test_data_batch_conversion.py:181:
  | _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  |  
  | data = array([list([1, 2, 3]), list([1])], dtype=object)
  | cast_tensor_columns = False
  |  
  | def _convert_batch_type_to_pandas(
  | data: DataBatchType,
  | cast_tensor_columns: bool = False,
  | ) -> "pd.DataFrame":
  | """Convert the provided data to a Pandas DataFrame.
  |  
  | Args:
  | data: Data of type DataBatchType
  | cast_tensor_columns: Whether tensor columns should be cast to NumPy ndarrays.
  |  
  | Returns:
  | A pandas Dataframe representation of the input data.
  |  
  | """
  | pd = _lazy_import_pandas()
  |  
  | if isinstance(data, dict):
  | tensor_dict = {}
  | for col_name, col in data.items():
  | if not isinstance(col, np.ndarray):
  | raise ValueError(
  | "All values in the provided dict must be of type "
  | f"np.ndarray. Found type {type(col)} for key {col_name} "
  | f"instead."
  | )
  | tensor_dict[col_name] = _ndarray_to_column(col)
  | data = pd.DataFrame(tensor_dict)
  | elif pyarrow is not None and isinstance(data, pyarrow.Table):
  | data = data.to_pandas()
  | elif not isinstance(data, pd.DataFrame):
  | >           raise ValueError(
  | f"Received data of type: {type(data)}, but expected it to be one "
  | f"of {DataBatchType}"
  | )
  | E           ValueError: Received data of type: <class 'numpy.ndarray'>, but expected it to be one of typing.Union[ForwardRef('numpy.ndarray'), ForwardRef('pandas.DataFrame'), typing.Dict[str, ForwardRef('numpy.ndarray')]]
```

Does that make sense? Anything I am misunderstanding here @Bhav00? Thanks!
Also cc @bveeramani for review + merge as well.

**Comment by Bhav00** [comment]
> I see, but the CI is still failing because those tests are erroring right? So if I understand correctly, either we need to take steps to fully remove the tests, or fix the errors such as [this one](https://buildkite.com/ray-project/oss-ci-build-pr/builds/30703#0189cbfc-3ce9-49bc-8124-b108df356444/2016-3075):
> 
> ```
> ___________________________ test_numpy_object_pandas ___________________________
> --
>   |  
>   | def test_numpy_object_pandas():
>   | input_data = np.array([[1, 2, 3], [1]], dtype=object)
>   | expected_output = pd.DataFrame({TENSOR_COLUMN_NAME: input_data})
>   | >       actual_output = _convert_batch_type_to_pandas(input_data)
>   |  
>   | python/ray/air/tests/test_data_batch_conversion.py:181:
>   | _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
>   |  
>   | data = array([list([1, 2, 3]), list([1])], dtype=object)
>   | cast_tensor_columns = False
>   |  
>   | def _convert_batch_type_to_pandas(
>   | data: DataBatchType,
>   | cast_tensor_columns: bool = False,
>   | ) -> "pd.DataFrame":
>   | """Convert the provided data to a Pandas DataFrame.
>   |  
>   | Args:
>   | data: Data of type DataBatchType
>   | cast_tensor_columns: Whether tensor columns should be cast to NumPy ndarrays.
>   |  
>   | Returns:
>   | A pandas Dataframe representation of the input data.
>   |  
>   | """
>   | pd = _lazy_import_pandas()
>   |  
>   | if isinstance(data, dict):
>   | tensor_dict = {}
>   | for col_name, col in data.items():
>   | if not isinstance(col, np.ndarray):
>   | raise ValueError(
>   | "All values in the provided dict must be of type "
>   | f"np.ndarray. Found type {type(col)} for key {col_name} "
>   | f"instead."
>   | )
>   | tensor_dict[col_name] = _ndarray_to_column(col)
>   | data = pd.DataFrame(tensor_dict)
>   | elif pyarrow is not None and isinstance(data, pyarrow.Table):
>   | data = data.to_pandas()
>   | elif not isinstance(data, pd.DataFrame):
>   | >           raise ValueError(
>   | f"Received data of type: {type(data)}, but expected it to be one "
>   | f"of {DataBatchType}"
>   | )
>   | E           ValueError: Received data of type: <class 'numpy.ndarray'>, but expected it to be one of typing.Union[ForwardRef('numpy.ndarray'), ForwardRef('pandas.DataFrame'), typing.Dict[str, ForwardRef('numpy.ndarray')]]
> ```
> 
> Does that make sense? Anything I am misunderstanding here @Bhav00? Thanks! Also cc @bveeramani for review + merge as well.

Similar to this one, the tests that use `data_batch_conversion.py`'s `_convert_batch_type_to_pandas` function will end up resulting in a probable error as this function had use of  `TENSOR_COLUMN_NAME` in the very first if condition which I removed. 
And maybe the other functions that used `TENSOR_COLUMN_NAME` will end up resulting in an error as long as the test case requires the use of '__value__'.
@scottjlee @bveeramani

**Comment by scottjlee** [comment]
I think you will need to modify the test to avoid passing in a standalone `np.ndarray`, because otherwise you will run into the type error as you mention above. The easiest way is to probably replace usage of `np.ndarray` with `Dict[str, np.ndarray]`; for example:
```
def test_numpy_pandas(cast_tensor_columns):
    input_data = {"data": np.array([1, 2, 3])}
    expected_output = pd.DataFrame(input_data)
    actual_output = _convert_batch_type_to_pandas(input_data, cast_tensor_columns)
    pd.testing.assert_frame_equal(expected_output, actual_output)

    output_array = _convert_pandas_to_batch_type(
        actual_output, type=BatchFormat.NUMPY, cast_tensor_columns=cast_tensor_columns
    )
    np.testing.assert_equal(output_array, input_data["data"])
```

File: python/ray/train/huggingface/transformers/transformers_predictor.py [hunk 1]
```diff
@@ -282,12 +281,6 @@ def _predict_pandas(
         feature_columns: Optional[List[str]] = None,
         **pipeline_call_kwargs,
     ) -> "pd.DataFrame":
-        if TENSOR_COLUMN_NAME in data:
-            arr = data[TENSOR_COLUMN_NAME].to_numpy()
-            if feature_columns:
-                data = pd.DataFrame(arr[:, feature_columns])
-        elif feature_columns:
-            data = data[feature_columns]
```
Comment (bveeramani) [inline python/ray/train/huggingface/transformers/transformers_predictor.py:290]: Do we still need this?

Comment (Bhav00) [reply python/ray/train/huggingface/transformers/transformers_predictor.py:290]: The line right below it says 
>      data = data[feature_columns] if feature_columns else data

both result in the same thing, so no I don't think we need it.

**Comment by Bhav00** [comment]
@scottjlee so changing from this code from the files
>    ``` np.testing.assert_equal(output_array, input_data) ```

to

>    ``` np.testing.assert_equal(output_array, input_data[TENSOR_COLUMN_NAME]) ```

should do it right? because the rest is just explicitly using `"data"` in the place of `TENSOR_COLUMN_NAME`

**Comment by Bhav00** [comment]
One thing I came to notice from multiple tests that failed is that it isn't what the test is asserting but what `_convert_batch_type_to_pandas()` is returning for the test to work with. 
The very first if, of the whole thing in that function that I removed was 
> ```    
>    if isinstance(data, np.ndarray):
>            data = pd.DataFrame({TENSOR_COLUMN_NAME: _ndarray_to_column(data)})
> ```   

It is doing exactly what you suggested to be done with the tests. 
I will be reverting the changes made to the three tests and instead, bring this 'if' back to this function to see what it achieves as soon as I get the approval.
@scottjlee

**Pull request closed without merging**